### PR TITLE
Light Node feature added.

### PR DESCRIPTION
**Pull request by alexeykiselev**

### CODE REVIEW

**Review by nickeskov — APPROVED** [review]

**Review by nickeskov — DISMISSED** [review]

File: pkg/state/transaction_checker.go [hunk 1]
```diff
@@ -88,23 +88,41 @@ func (tc *transactionChecker) scriptActivation(libVersion ast.LibraryVersion, ha
 	if err != nil {
 		return scriptFeaturesActivations{}, err
 	}
+	lightNodeActivated, err := tc.stor.features.newestIsActivated(int16(settings.LightNode))
+	if err != nil {
+		return scriptFeaturesActivations{}, err
+	}
+	if libVersion < ast.LibV4 && lightNodeActivated {
+		return scriptFeaturesActivations{},
+			errors.New("scripts versions 1, 2 and 3 are disable after activation of Light Node feature")
```
Comment (nickeskov) [inline pkg/state/transaction_checker.go:97]: Maybe change message to this:
```go
errors.Errorf("scripts with versions below %d are disabled after activation of Light Node feature", ast.LibV4)
```

File: pkg/ride/functions_list.go [hunk 2]
```diff
@@ -451,18 +457,27 @@ func listRemoveByIndex(_ environment, args ...rideType) (rideType, error) {
...
+	if err != nil {
+		return nil, errors.Wrap(err, "listReplaceByIndex")
+	}
+	l := len(list)
+	if l == 0 {
+		return nil, errors.New("listReplaceByIndex: can't replace an element in the empty list")
+	}
+	r := make(rideList, l)
+	copy(r, list)
+	r[i] = el
```
[the remark this answers is not in the document]
Comment (alexeykiselev) [reply pkg/ride/functions_list.go:477]: Yes, it was made intendedly. All list functions operates on copies of lists.

File: pkg/settings/features.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ const (
 	BlockRewardDistribution // 19
 	CappedRewards           // 20
 	XTNBuyBackCessation     // 21
+	LightNode               // 22
 	InvokeExpression        // 22
```
[the remark this answers is not in the document]
Comment (alexeykiselev) [reply pkg/settings/features.go:28]: Ok, fixed.

File: pkg/state/transaction_checker.go (hunk 1 above)
Comment (alexeykiselev) [reply pkg/state/transaction_checker.go:97]: Yes, that sounds better. Replaced.

**Review by nickeskov — APPROVED** [review]

**Pull request merged**